mifosng-android/src/main/java/com/mifos/mifosxdroid/online/generatecollectionsheet/GenerateCollectionSheetPresenter.java

```diff
@@ -155,15 +222,356 @@ public void onCompleted() {
+                        getMvpView().showCollection(sheet);
+                    }
+                }));
+    }
+
+    /**
+     * Method to submit the ProductiveCollectionSheet
+     */
+    public void submitProductiveSheet(int centerId, ProductiveCollectionSheetPayload payload) {
+        checkViewAttached();
+        getMvpView().showProgressbar(true);
+        mSubscription.add(collectionDataManager.submitProductiveSheet(centerId, payload)
+                .observeOn(AndroidSchedulers.mainThread())
+                .subscribeOn(Schedulers.io())
+                .subscribe(new Subscriber<GenericResponse>() {
+                    @Override
+                    public void onCompleted() {
+
+                    }
+
+                    @Override
+                    public void onError(Throwable e) {
+                        getMvpView().showProgressbar(false);
+                        try {
+                            if (e instanceof HttpException) {
+                                String errorMessage = ((HttpException) e).response().errorBody()
+                                        .string();
+                                if (((HttpException) e).code() == 500) {
+                                    getMvpView().showError(errorMessage);
+                                } else {
+                                    getMvpView().showError(MFErrorParser.parseError(errorMessage)
+                                            .getErrors().get(0).getDefaultUserMessage());
+                                }
+                            }
+                        } catch (Throwable throwable) {
+                            RxJavaPlugins.getInstance().getErrorHandler().handleError(e);
+                        }
+                    }
+
+                    @Override
+                    public void onNext(GenericResponse response) {
+                        getMvpView().showProgressbar(false);
+                        getMvpView().showError(c.getResources().getString(
+                                R.string.collectionsheet_submit_success));
+                    }
+                }));
+    }
+
+    /**
+     * Method to submit the CollectionSheet
+     */
+    public void submitCollectionSheet(int groupId, CollectionSheetPayload payload) {
+        checkViewAttached();
+        getMvpView().showProgressbar(true);
+        mSubscription.add(collectionDataManager.submitCollectionSheet(groupId, payload)
+                .observeOn(AndroidSchedulers.mainThread())
+                .subscribeOn(Schedulers.io())
+                .subscribe(new Subscriber<GenericResponse>() {
+                    @Override
+                    public void onCompleted() {
+
+                    }
+
+                    @Override
+                    public void onError(Throwable e) {
+                        getMvpView().showProgressbar(false);
+                        try {
+                            if (e instanceof HttpException) {
```

Comment:
use this utils method to show error https://github.com/openMF/android-client/blob/master/mifosng-android/src/main/java/com/mifos/utils/MFErrorParser.java#L25